substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/JavaLangSubstitutions.java

```diff
@@ -758,15 +758,20 @@ final class Target_jdk_internal_loader_ClassLoaders {
 @TargetClass(value = jdk.internal.loader.BootLoader.class)
 final class Target_jdk_internal_loader_BootLoader {
 
+    /**
+     * Returns the Package of the given name defined to the boot loader or null if the package has
+     * not been defined.
+     */
     @Substitute
-    static Package getDefinedPackage(String name) {
+    public static Package getDefinedPackage(String name) {
         if (name != null) {
-            Target_java_lang_Package pkg = new Target_java_lang_Package(name, null, null, null,
-                            null, null, null, null, null);
-            return SubstrateUtil.cast(pkg, Package.class);
-        } else {
-            return null;
+            for (Package pkg : (Iterable<Package>) packages()::iterator) {
```

Comment:
Why convert the `Stream` to an `Iteraable` instead of using proper stream expressions (`filter`, `empty`)?

Reply:
Because it seemed simpler.